x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+
+    assert!(
+        BURN_AMOUNT <= balance,
+        "amount burned must be less than or equal to the user balance"
+    );
+
+    let counter = program
+        .state()
+        .get::<i64, _>(StateKey::Counter.to_vec())
+        .unwrap_or_default();
+
+    assert!(counter > 0, "cannot burn more nfts");
+
+    // Burn the NFT by transferring it to the zero address
+    program
+        .state()
+        .store(StateKey::Balance(from).to_vec(), &(balance - BURN_AMOUNT))
+        .expect("failed to store new balance");
+
+    // TODO move to a lazy static? Or move to the VM layer entirely
+    program
+        .state()
+        .store(StateKey::Owner.to_vec(), &null_address)
+        .is_ok()
+}
+
+#[cfg(test)]
+mod tests {
+    use serial_test::serial;
+    use std::env;
+    use wasmlanche_sdk::simulator::{self, Key, PlanResponse};
+
+    use crate::example;
+
+    // export SIMULATOR_PATH=/path/to/simulator
+    // export PROGRAM_PATH=/path/to/program.wasm
+    // cargo cargo test --package token --lib nocapture -- tests::test_token_plan --exact --nocapture --ignored
+    #[test]
+    #[serial]
+    #[ignore = "requires SIMULATOR_PATH and PROGRAM_PATH to be set"]
+    fn test_single_nft_plan() {
+        use wasmlanche_sdk::simulator::{self, Key};
+        let s_path = env::var(simulator::PATH_KEY).expect("SIMULATOR_PATH not set");
+        let simulator = simulator::Client::new(s_path);
+
+        let alice_key = "alice_key";
+        // create owner key in single step
+        let resp = simulator
+            .key_create::<PlanResponse>(alice_key, Key::Ed25519)
+            .unwrap();
+        assert_eq!(resp.error, None);
+
+        // create multiple step test plan
+        let nft_name = "MyNFT";
+        let binding = nft_name.len().to_string();
+        let nft_name_length: &str = binding.as_ref();
+
+        let nft_symbol = "MNFT";
+        let binding = nft_symbol.len().to_string();
+        let nft_symbol_length: &str = binding.as_ref();
+
+        let nft_uri = "ipfs://my-nft.jpg";
+        let binding = nft_uri.len().to_string();
+        let nft_uri_length: &str = binding.as_ref();
+
+        let plan = example::initialize_plan(
+            nft_name,
+            nft_name_length,
+            nft_symbol,
+            nft_symbol_length,
+            nft_uri,
+            nft_uri_length,
+        );
+
+        // run plan
+        let plan_responses = simulator.run::<PlanResponse>(&plan).unwrap();
+
+        // collect actual id of program from step 0
+        let mut program_id = String::new();
+        if let Some(step_0) = plan_responses.first() {
+            program_id = step_0.result.id.clone().unwrap_or_default();
+        }
+
+        // ensure no errors
+        assert!(
+            plan_responses.iter().all(|resp| resp.error.is_none()),
+            "error: {:?}",
+            plan_responses
+                .iter()
+                .filter_map(|resp| resp.error.as_ref())
+                .next()
+        );
```

Comment:
Should you print all the errors instead of just the first one?

Reply:
Please reply to questions before resolving comments.

Reply:
Any error is terminal